scripts/ci/check-encoding.mjs

```
@@ -0,0 +1,117 @@
+ * @fileoverview UTF-8 encoding validation for source files
+ */
+
+import { readFileSync, statSync } from 'node:fs';
+import { basename } from 'node:path';
+
+// UTF-8 BOM bytes
+const UTF8_BOM = Buffer.from([0xef, 0xbb, 0xbf]);
+
+// Invalid characters to check for (common encoding issues)
+const INVALID_CHAR_PATTERNS = [
+  { pattern: /[\x00-\x08\x0b\x0c\x0e-\x1f]/, name: 'control characters' },
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)